lib/core/macros.nim

```diff
@@ -856,7 +856,9 @@ proc nestList*(op: NimNode; pack: NimNode; init: NimNode): NimNode =
   for i in countdown(pack.len - 1, 0):
     result = newCall(op, pack[i], result)
 
-proc treeTraverse(n: NimNode; res: var string; level = 0; isLisp = false, indented = false) {.benign.} =
+const collapseSymChoiceDefault = not defined(nimLegacyMacrosCollapseSymChoice)
+
+proc treeTraverse(n: NimNode; res: var string; level = 0; isLisp = false, indented = false, collapseSymChoice = collapseSymChoiceDefault) {.benign.} =
```

Comment:
I don't understand why this needs a default parameter, why not just use the `collapseSymChoiceDefault` constant?

Reply:
so that user can call `treeRepr(n, collapseSymChoice = true|false)` if they choose instead of relying on global `nimLegacyMacrosCollapseSymChoice`;

in future, an additional optional param `renderSymCallback` could be passed to customize how symbols are rendered (a callback being the most flexible, eg to show declaredloc for symbols).

Here though, `collapseSymChoice = false` behavior wasn't useful so I'd ok to not expose this flag to users (and just use collapseSymChoiceDefault internally as you suggest) if you insist

Reply:
I do insist.

Reply:
done